AppCenterData/AppCenterData/MSData.m

```diff
@@ -753,12 +753,18 @@ - (void)authTokenContext:(MSAuthTokenContext *)__unused authTokenContext didUpda
   // If user logs in.
   if (userInfomation && userInfomation) {
     [self.dataOperationProxy.documentStore createUserStorageWithAccountId:userInfomation.accountId];
+
+    // Persist Auth context availability flag.
+    [MS_USER_DEFAULTS setObject:@YES forKey:kMSAuthContextAvailabilityKey];
   } else {
     // If user logs out.
     [MSTokenExchange removeAllCachedTokens];
 
     // Delete all the data (user and read-only).
     [self.dataOperationProxy.documentStore resetDatabase];
+
+    // Clean Auth context availability flag.
+    [MS_USER_DEFAULTS removeObjectForKey:kMSAuthContextAvailabilityKey];
```

Comment:
Since you are removing the value for the key here, it should be added to a method when Data is disabled.